processing/src/main/java/org/apache/druid/segment/ColumnProcessors.java

```diff
@@ -342,7 +342,7 @@ private static ColumnCapabilities computeDimensionSpecCapabilities(
           );
         }
 
-        if (mayBeMultiValue(capabilities)) {
+        if (capabilities.hasMultipleValues().isMaybeTrue()) {
```

Comment:
nit: perhaps it would be useful to add a util method like `shouldProcessMultiValue()` that can be shared by here and `StringAnyAggregatorFactory`.

Reply:
NVM. I misread the change.

Reply:
there isn't really anything to be shared here, `StringAnyAggregatorFactory` needs a null check (different from this shared method since it is only applicable to non-vectorized engines) while this part of the code has already checked null, so I think the shared method would only really be used by `StringAnyAggregatorFactory`.